app/ConfigRepository.php

```diff
@@ -106,6 +113,12 @@ public function get($key, $default = null): mixed
             return $default;
         }
 
+        // Check if we need to load the OS YAML file
+        if (str_starts_with($key, 'os.')) {
+            $os = substr($key, 3, (strpos($key, '.', 3) ?: strlen($key)) - 3);
```

Comment:
Dang, that is a bit more convoluted than I thought.  It might be better to do preg_match after doing the str_starts_with.  It might be a little odd, but I do want to be mindful of performance in this code as it is called a lot.

Reply:
The other option is:
$os = explode('.', $key)[1];

Is preg_match still preferred over explode()?